File: image.m
pkg load image  % Load the image package if not already loaded

% Read image file
image = imread('input_image.jpg');  % Replace 'input_image.jpg' with the path to your image file

% Convert the image to RGB (assuming it's in RGB format)
if size(image, 3) == 1
    rgb_image = cat(3, image, image, image);  % Convert grayscale to RGB
else
    rgb_image = image;  % Image is already in RGB format
end

% Get RGB values as a 2D matrix
[m, n, ~] = size(rgb_image);
rgb_values = reshape(rgb_image, m*n, 3);

% Write RGB values to CSV file
csvwrite('rgb_values.csv', rgb_values);

File: image.py
from PIL import Image
import numpy as np

# Read image file
image = Image.open('input_image.jpg')  # Replace 'input_image.jpg' with the path to your image file
rgb_image = image.convert('RGB')

# Get RGB values as a numpy array
rgb_array = np.array(rgb_image)

# Reshape array to a 2D array (rows: pixels, columns: RGB values)
pixels, channels = rgb_array.shape[:-1]
rgb_values = rgb_array.reshape(pixels, channels)

# Write RGB values to CSV file
np.savetxt('rgb_values.csv', rgb_values, delimiter=',')
